### PR TITLE
Refactor: Clean up EAGLE training dataset preparation

### PR DESCRIPTION
**Pull request by benchislett**
## What does this PR do?

**Type of change:** Refactor

**Overview:** 
- Consolidate input dataset preparation into `make_dataset.py`
- Read dataset mix spec from a YAML file
- - Can now specify how many samples to take from each split
- - Can no longer easily split a dataset into train/test sections. I don't think this feature was really useful to begin with. Most datasets can already be separated into train/val/test at the split level, and those that can't are usually going to be splitted by the training FW anyways.
- Add support for a few new dataset types, magpie 300k/500k/1M, nemotron post-training dataset v2.

## Usage
See README for detailed example

## Testing
Ran it locally on all dataset modes, works successfully and output looks good. Checked shuffling, conversation IDs, and output contents were all unique and usable.

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes
- **Did you write any new necessary tests?**: No
- **Did you add or update any necessary documentation?**: Yes
- **Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?**: No 

## Summary by CodeRabbit

* **Documentation**
  * Updated speculative decoding example documentation with new dataset references and standardized file paths.

* **New Features**
  * Introduced configuration-driven dataset preparation supporting multiple dataset sources with centralized configuration files.

* **Refactor**
  * Simplified dataset preparation workflow with unified tooling and updated default data paths throughout the training pipeline.

### CODE REVIEW

File: examples/speculative_decoding/prepare_input_conversations/example_data_config.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+outputs:
+  - filename: "input_conversations/train.jsonl"
+    global_limit: 50000
+    sources:
+      - name: "sharegpt"
+        splits:
+          all: 0
```
Comment (yeyu-nvidia) [inline examples/speculative_decoding/prepare_input_conversations/example_data_config.yaml:7]: what does 0 mean

**Comment by yeyu-nvidia** [comment]
Can you rebase to main as there are some conflicts?

File: examples/speculative_decoding/prepare_input_conversations/example_data_config.yaml (hunk 1 above)
Comment (benchislett) [reply examples/speculative_decoding/prepare_input_conversations/example_data_config.yaml:7]: 0 means "no samples". This is just an example template file showing which datasets are supported, so include most of the options even if they aren't used by default
